lib/eth/client.rb

```diff
@@ -261,21 +261,6 @@ def call_raw(contract, func, *args, **kwargs)
           max_gas_fee: max_fee_per_gas,
         })
       end
-      unless kwargs[:sender_key].nil?
-        # use the provided key as sender and signer
-        params.merge!({
-          from: kwargs[:sender_key].address,
-          nonce: get_nonce(kwargs[:sender_key].address),
-        })
-        tx = Eth::Tx.new(params)
-        tx.sign kwargs[:sender_key]
-      else
```

Comment:
Or we do an `elsif` here checking if the coinbase is available and catching the error code in case it's not.